quipucords/scanner/network/inspect_callback.py

```diff
@@ -46,30 +71,37 @@ def __init__(self, scan_task, inspect_results, display=None):
         self.results = []
         self._ansible_facts = {}
 
+    # Ansible considers tasks failed when their return code is
+    # nonzero, even if we set ignore_errors=True. We want to be able
+    # to process those task results, so we need to handle failed
+    # results the same way we treat ok ones.
     def v2_runner_on_ok(self, result):
         """Print a json representation of the result."""
+        self.handle_result(result)
+
+    def v2_runner_on_failed(self, result, ignore_errors=False):
+        """Print a json representation of the result."""
+        self.handle_result(result)
+
+    def handle_result(self, result):
+        """Handle an incoming result object."""
         result_obj = _construct_result(result)
         self.results.append(result_obj)
         logger.debug('%s', result_obj)
-        # pylint: disable=protected-access
-        host = str(result._host)
-        if (result._result is not None and isinstance(result._result, dict) and
-                'ansible_facts' in result._result):
-            host_facts = result._result['ansible_facts']
-            facts = {}
-            for key, value in host_facts.items():
-                if key == 'host_done':
-                    self._finalize_host(host)
-                elif not key.startswith('internal'):
-                    facts[key] = value
-                # Facts starting with 'internal' are used only by the
-                # Ansible playbooks to compute other
-                # facts. Deliberately drop them here.
-
-            if host in self._ansible_facts:
-                self._ansible_facts[host].update(facts)
+
+        host = result_obj['host']
+        results_to_store = normalize_result(result)
+        host_facts = {}
+        for key, value in results_to_store:
+            if key == 'host_done':
```

Comment:
Can we make 'host_done' a constant? I know you didn't change this but it would be good for consistency and memory allocation.

Reply:
no problem, done!